From: Sergey Senozhatsky <senozhatsky@chromium.org>
Subject: zsmalloc: break out of loop when found perfect zspage order
Date: Tue, 25 Oct 2022 01:12:12 +0900

If we found zspage configuration that gives us perfect 100% used
percentage (zero wasted space) then there is no point it trying any other
configuration

Link: https://lkml.kernel.org/r/20221024161213.3221725-6-senozhatsky@chromium.org
Signed-off-by: Sergey Senozhatsky <senozhatsky@chromium.org>
Cc: Minchan Kim <minchan@kernel.org>
Cc: Nitin Gupta <ngupta@vflare.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/zsmalloc.c |    3 +++
 1 file changed, 3 insertions(+)

--- a/mm/zsmalloc.c~zsmalloc-break-out-of-loop-when-found-perfect-zspage-order
+++ a/mm/zsmalloc.c
@@ -790,6 +790,9 @@ static int get_pages_per_zspage(struct z
 			max_usedpc = usedpc;
 			max_usedpc_order = i;
 		}
+
+		if (usedpc == 100)
+			break;
 	}
 
 	return max_usedpc_order;
_
